### ISSUE TITLE
[P2] aligned_metas 只比 mask，會把後一次碰牌 logits 配給先前的 pass，產生沒有改變動作的假偏離

### ISSUE THREAD

**Issue by HHim8826** [HHim8826/Mortal#12]
審查分支 `train-parquet`，版本 `f228ffffb7fab60115507ee68b32c1adba8d550e`。由 `f228fff` 新增的反事實評估實際使用到這條路徑；相同 cursor 邏輯原已存在於 behaviour_of。

[rollout.py:184–194](https://github.com/HHim8826/Mortal/blob/f228ffffb7fab60115507ee68b32c1adba8d550e/mortal/rollout.py#L184-L194)；[deviation_cost.py:205–237](https://github.com/HHim8826/Mortal/blob/f228ffffb7fab60115507ee68b32c1adba8d550e/mortal/deviation_cost.py#L205-L237)；[相同牌譜仍收為觀測：355–371](https://github.com/HHim8826/Mortal/blob/f228ffffb7fab60115507ee68b32c1adba8d550e/mortal/deviation_cost.py#L355-L371)。

放過副露沒有 mjai event/meta，下一份 metadata 可能來自更晚的決策。兩次決策可以有完全相同的 legal-action mask，例如先放過白板的碰、稍後接受發財的碰，兩次都是 {41,45}。aligned_metas 只比較 mask_bits，於是先前的 pass 吃掉後一次 pon 的 logits；真正的 pon 反而得到 None。

已用本機原生 libriichi.mjai.Bot 產生上述 pass／pon／discard 序列，再交給真正 GameplayLoader 解碼，不只是手工輸入 actions：
```text
decoded actions:                 [45, 41, 0]
前兩個 masks:                    [[41,45], [41,45]]
應該附有 metadata:               [False, True, True]
實際 aligned_metas:              [True, False, True]
第一筆被錯配的 q_values:         [2.0, 0.0]  # 原本屬於 pon

pick_target(..., rng seed=3):
index=0, claimed argmax=41, forced=45
但 baseline 在 index=0 本來就是 45。
```

這會把 pass→pass 當成一次「離開 argmax」的實驗。Forcer 可以成功命中正確 observation，但命中只證明找對狀態，不能證明配給這個狀態的 logits 正確。main 對 forked_at=None 只寫 debug log，仍把 0 效果放進統計；目標抽樣的權重也已用錯分布。因而 witness 全部相同也無法揭露這種污染。

建議：依事件／決策位置建立對齊，不能以 mask 相同視為同一決策；對原本不保留 metadata 的 pass／結束局動作，可從對應 observation 重新計算 logits。對 argmax baseline 額外驗證選出的 argmax 等於該 instance 的 actual action，以及 forced 確實不同，錯配時拒絕該觀測。

**Comment by HHim8826** [comment]
Fixed in 0833528.

The baseline plays its argmax, so the logits a target was picked from have to name the action the baseline actually played. A target where they disagree is now dropped and counted, which turns `aligned_metas` mis-assigning a pass the logits of a later pon into a dropped observation rather than a fake deviation. Measured over the 900 forced deviations of the run this came out of: none disagreed.

The deeper alignment repair you suggested -- indexing by event position rather than by mask, and recomputing logits for pass and kyoku-ending actions -- is not done. `aligned_metas` still walks the cursor `behaviour_of` walks, so the mis-assignment can still happen; what changed is that it can no longer reach the statistics silently. Worth noting that `test_rollout`'s `test_every_decision_is_accounted_for` turns out to be non-deterministic -- the arena advances four hanchans at once and the sampling order varies, so it plays a different game each run -- and some of those games do produce an 'unknown'. The alignment is not exact on real logs, so a rate measured over many games would be the right next step rather than an assertion on one.

**Comment by HHim8826** [comment]
Confirmed in the field, one block after the guard went in:

```
100481_53759_a.json.gz: the metadata says the argmax here is 41 and the
policy played 45, so this decision was paired with the logits of another one
```

Action 41 is the pon, 45 the pass — the pair you described, sharing the mask, with the pass taking the later pon's logits. Your repro was synthetic; this is the same thing on a real self-play log. The target was dropped rather than measured, so the rate is roughly 1 in 1,800 decisions picked this way, and the statistics never saw it.

That rate is the argument for not stopping here. A guard that drops the observation is right for a measurement, but the underlying alignment is still wrong wherever a pass precedes a same-mask call, and anything that reads `aligned_metas` or `behaviour_of` without a way to check itself — the importance ratio in `train_ppo`, which has no argmax to compare against — has no such guard. Recorded here rather than left in a log.
